static/src/javascripts/projects/common/modules/commercial/enhanced-consent.ts

```diff
@@ -0,0 +1,69 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type {
+	ConsentState,
+	Framework,
+} from '@guardian/consent-management-platform/dist/types';
+import type { TCFv2ConsentList } from '@guardian/consent-management-platform/dist/types/tcfv2';
+
+type ConsentStateEnhanced = ConsentState & {
+	canTarget: boolean;
+	framework: Framework | null;
+};
+
+const tcfv2ConsentedToAll = (consents: TCFv2ConsentList): boolean => {
+	return (
+		Object.keys(consents).length > 0 &&
+		Object.values(consents).every(Boolean)
+	);
+};
+
+const buildConsentStateEnhanced = (
+	consentState: ConsentState,
+	canTarget: boolean,
+	framework: Framework | null,
+): ConsentStateEnhanced => ({
+	...consentState,
+	canTarget,
+	framework,
+});
+
+const enhanceConsentState = (state: ConsentState): ConsentStateEnhanced => {
+	if (state.tcfv2) {
+		return buildConsentStateEnhanced(
+			state,
+			tcfv2ConsentedToAll(state.tcfv2.consents),
+			'tcfv2',
+		);
+	} else if (state.ccpa) {
+		return buildConsentStateEnhanced(state, !state.ccpa.doNotSell, 'ccpa');
+	} else if (state.aus) {
+		return buildConsentStateEnhanced(
+			state,
+			state.aus.personalisedAdvertising,
+			'aus',
+		);
+	}
+	return buildConsentStateEnhanced(state, false, null);
+};
+
+const getEnhancedConsent = (): Promise<ConsentStateEnhanced> =>
+	new Promise<ConsentStateEnhanced>((resolve, reject) => {
+		onConsentChange((consentState) => {
+			if (consentState.tcfv2) {
+				// For tcfv2 only, the first onConsentChange is fired before the user has
+				// interacted with the consent banner. We want to ignore this first consent state.
+				if (consentState.tcfv2.eventStatus !== 'cmpuishown') {
+					resolve(enhanceConsentState(consentState));
+				}
+				return;
+			} else if (consentState.ccpa || consentState.aus) {
+				resolve(enhanceConsentState(consentState));
+			}
+
+			reject('Unknown framework');
+		});
+	});
```

Comment:
I’d be tempted to have this depend on `getInitialConsentState`…

I understand that technically, if you call this after a further consent signal has been sent, it may not be the “initial” state, so happy to be _pushed_ or _pulled_ on this one

Reply:
It was reverted in v10, sadly :(

Reply:
@zekehuntergreen It would allow us to tidy up our promises around `onConsentChange` quite a bit. Is it possible / advisable to add the change independently to CMP and release that as v11?

Reply:
Yes since it's a non-breaking change we could add this back in as a bug fix like the original PR and release it as v10.1.2